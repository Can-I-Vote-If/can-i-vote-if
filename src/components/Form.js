import React, {
  Component
} from 'react';

import States from './States';
import Results from './Results';

class Form extends Component {
  constructor(props) {
    super(props);
    this.state = {
      indState: null,
      citizen: null,
      age: null,
      crimes: null,
      submitted: null
    };
  }

  handleAgeChange = (event) => {
    let dob = event.target.value;
    let today = new Date();
    let birthDate = new Date(dob);
    let age = today.getFullYear() - birthDate.getFullYear();
    let m = today.getMonth() - birthDate.getMonth();
    if (m < 0 || (m === 0 && today.getDate() < birthDate.getDate())) {
      age--;
    }

    this.setState({
      age: age
    });
  }

  handleChange = (e, name) => {
    const value = e.target.value;
    this.setState({
      [name]: value
    });
  }

  handleSubmit = (event) => {
    event.preventDefault();
    this.setState({
      submitted: true
    });
  }

  render() {
    return (
      <form id="form" onSubmit={this.handleSubmit}>
        <div className="row">
          <div className="col-lg-11 col-lg-offset-1">
            <h3 className="subHead">Basic Information</h3>
          </div>
        </div>
        <div className="row">
          <div className="col-lg-2 col-lg-offset-1">
            <States action={e => this.handleChange(e, 'indState')} />
          </div>

          <div className="col-lg-2">
            <div className="radio-group form-group field-us-citizen">
              <label htmlFor="us-citizen" className="radio-group-label">
                Are you a U.S. Citizen?<span className="required">*</span>
              </label>
              <br />
              <div className="btn-group" role="group" aria-label="...">
                <input
                  type="button"
                  className={
                    this.state.citizen === 'Yes'
                      ? 'btn btn-default isClicked'
                      : 'btn btn-default'
                  }
                  name="us-citizen"
                  id="us-citizen-0"
                  required="required"
                  value="Yes"
                  onClick={e => this.handleChange(e, 'citizen')}
                />
                <input
                  type="button"
                  className={
                    this.state.citizen === 'No'
                      ? 'btn btn-default isClicked'
                      : 'btn btn-default'
                  }
                  name="us-citizen"
                  id="us-citizen-1"
                  required="required"
                  value="No"
                  onClick={e => this.handleChange(e, 'citizen')}
                />
              </div>
            </div>
          </div>

          <div className="col-lg-2">
            <div className="date form-group field-birthdate">
              <label htmlFor="birthdate" className="date-label">
                Birthday<span className="required">*</span>
              </label>
              <input
                type="date"
                className="form-control"
                name="birthdate"
                id="birthdate"
                required="required"
                aria-required="true"
                onChange={this.handleAgeChange}
              />
            </div>
          </div>

          <div className="col-lg-5">
            <div className="radio-group form-group field-us-citizen">
              <label htmlFor="us-citizen" className="radio-group-label">
                Have you ever been charged, or convicted of a crime?<span className="required">
                  *
                </span>
              </label>
              <br />
              <div className="btn-group" role="group" aria-label="...">
                <input
                  type="button"
                  className={
                    this.state.crimes === 'Yes'
                      ? 'btn btn-default isClicked'
                      : 'btn btn-default'
                  }
                  name="us-crimes"
                  id="us-crimes-0"
                  required="required"
                  value="Yes"
                  onClick={e => this.handleChange(e, 'crimes')}
                />
                <input
                  type="button"
                  className={
                    this.state.crimes === 'No'
                      ? 'btn btn-default isClicked'
                      : 'btn btn-default'
                  }
                  name="us-crimes"
                  id="us-crimes-1"
                  required="required"
                  value="No"
                  onClick={e => this.handleChange(e, 'crimes')}
                />
              </div>
            </div>
          </div>

          <div className="col-sm-12 parentButton">
            <input
              id="submit"
              className="btn btn-primary"
              type="submit"
              value="View Your Eligibility"
            />
          </div>
        </div>

        {this.state.submitted ? (
          <Results
            age={this.state.age}
            citizen={this.state.citizen}
            crimes={this.state.crimes}
            state={this.state.indState}
          />
        ) : null}
      </form>
    );
  }
}

export default Form;
