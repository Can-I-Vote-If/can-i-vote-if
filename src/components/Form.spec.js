import React from "react";
import renderer from "react-test-renderer";
import { mount } from "enzyme";

import Form from "./Form";

describe("components/Form", async () => {
  it("should render", async () => {
    const tree = renderer.create(<Form />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  // This might work.  Need to find a workaround for setting up enzyme with create-react-app
  /* it("should view eligibility if all fields are entered", async () => {
   *   const wrapper = mount(<Form />);
   *   wrapper.setState({
   *     state: "Minnesota",
   *     citizen: "Yes",
   *     age: new Date("1995-07-04"),
   *     crimes: "No"
   *   });
   *   expect(wrapper.toJSON()).toMatchSnapshot();
   * }); */
});
