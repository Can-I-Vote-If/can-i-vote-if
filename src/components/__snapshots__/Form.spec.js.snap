// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Form should render 1`] = `
<form
  id="form"
  onSubmit={[Function]}
>
  <div
    className="row"
  >
    <div
      className="col-lg-11 col-lg-offset-1"
    >
      <h3
        className="subHead"
      >
        Basic Information
      </h3>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-lg-2 col-lg-offset-1"
    >
      <div
        className="select form-group field-state"
      >
        <label
          className="select-label"
          htmlFor="state"
        >
          Where do you live?
          <span
            className="required"
          >
            *
          </span>
        </label>
        <select
          aria-required="true"
          className="form-control"
          defaultValue="Select State"
          id="state"
          name="state"
          onChange={[Function]}
          required="required"
        >
          <option
            disabled={true}
            value="Select State"
          >
            Select State
          </option>
          <option
            value="Alabama"
          >
            Alabama
          </option>
          <option
            value="Alaska"
          >
            Alaska
          </option>
          <option
            value="Arizona"
          >
            Arizona
          </option>
          <option
            value="Arkansas"
          >
            Arkansas
          </option>
          <option
            value="California"
          >
            California
          </option>
          <option
            value="Colorado"
          >
            Colorado
          </option>
          <option
            value="Connecticut"
          >
            Connecticut
          </option>
          <option
            value="Delaware"
          >
            Delaware
          </option>
          <option
            value="District of Columbia"
          >
            District of Columbia
          </option>
          <option
            value="Florida"
          >
            Florida
          </option>
          <option
            value="Georgia"
          >
            Georgia
          </option>
          <option
            value="Hawaii"
          >
            Hawaii
          </option>
          <option
            value="Idaho"
          >
            Idaho
          </option>
          <option
            value="Illinois"
          >
            Illinois
          </option>
          <option
            value="Indiana"
          >
            Indiana
          </option>
          <option
            value="Iowa"
          >
            Iowa
          </option>
          <option
            value="Kansas"
          >
            Kansas
          </option>
          <option
            value="Kentucky"
          >
            Kentucky
          </option>
          <option
            value="Louisiana"
          >
            Louisiana
          </option>
          <option
            value="Maine"
          >
            Maine
          </option>
          <option
            value="Maryland"
          >
            Maryland
          </option>
          <option
            value="Massachusetts"
          >
            Massachusetts
          </option>
          <option
            value="Michigan"
          >
            Michigan
          </option>
          <option
            value="Minnesota"
          >
            Minnesota
          </option>
          <option
            value="Mississippi"
          >
            Mississippi
          </option>
          <option
            value="Missouri"
          >
            Missouri
          </option>
          <option
            value="Montana"
          >
            Montana
          </option>
          <option
            value="Nebraska"
          >
            Nebraska
          </option>
          <option
            value="Nevada"
          >
            Nevada
          </option>
          <option
            value="New Hampshire"
          >
            New Hampshire
          </option>
          <option
            value="New Jersey"
          >
            New Jersey
          </option>
          <option
            value="New Mexico"
          >
            New Mexico
          </option>
          <option
            value="New York"
          >
            New York
          </option>
          <option
            value="North Carolina"
          >
            North Carolina
          </option>
          <option
            value="North Dakota"
          >
            North Dakota
          </option>
          <option
            value="Ohio"
          >
            Ohio
          </option>
          <option
            value="Oklahoma"
          >
            Oklahoma
          </option>
          <option
            value="Oregon"
          >
            Oregon
          </option>
          <option
            value="Pennsylvania"
          >
            Pennsylvania
          </option>
          <option
            value="Rhode Island"
          >
            Rhode Island
          </option>
          <option
            value="South Carolina"
          >
            South Carolina
          </option>
          <option
            value="South Dakota"
          >
            South Dakota
          </option>
          <option
            value="Tennessee"
          >
            Tennessee
          </option>
          <option
            value="Texas"
          >
            Texas
          </option>
          <option
            value="Utah"
          >
            Utah
          </option>
          <option
            value="Vermont"
          >
            Vermont
          </option>
          <option
            value="Virginia"
          >
            Virginia
          </option>
          <option
            value="Washington"
          >
            Washington
          </option>
          <option
            value="West Virginia"
          >
            West Virginia
          </option>
          <option
            value="Wisconsin"
          >
            Wisconsin
          </option>
          <option
            value="Wyoming"
          >
            Wyoming
          </option>
        </select>
      </div>
    </div>
    <div
      className="col-lg-2"
    >
      <div
        className="radio-group form-group field-us-citizen"
      >
        <label
          className="radio-group-label"
          htmlFor="us-citizen"
        >
          Are you a U.S. Citizen?
          <span
            className="required"
          >
            *
          </span>
        </label>
        <br />
        <div
          aria-label="..."
          className="btn-group"
          role="group"
        >
          <input
            className="btn btn-default"
            id="us-citizen-0"
            name="us-citizen"
            onClick={[Function]}
            required="required"
            type="button"
            value="Yes"
          />
          <input
            className="btn btn-default"
            id="us-citizen-1"
            name="us-citizen"
            onClick={[Function]}
            required="required"
            type="button"
            value="No"
          />
        </div>
      </div>
    </div>
    <div
      className="col-lg-2"
    >
      <div
        className="date form-group field-birthdate"
      >
        <label
          className="date-label"
          htmlFor="birthdate"
        >
          Birthday
          <span
            className="required"
          >
            *
          </span>
        </label>
        <input
          aria-required="true"
          className="form-control"
          id="birthdate"
          name="birthdate"
          onChange={[Function]}
          required="required"
          type="date"
        />
      </div>
    </div>
    <div
      className="col-lg-5"
    >
      <div
        className="radio-group form-group field-us-citizen"
      >
        <label
          className="radio-group-label"
          htmlFor="us-citizen"
        >
          Have you ever been charged, or convicted of a crime?
          <span
            className="required"
          >
            *
          </span>
        </label>
        <br />
        <div
          aria-label="..."
          className="btn-group"
          role="group"
        >
          <input
            className="btn btn-default"
            id="us-crimes-0"
            name="us-crimes"
            onClick={[Function]}
            required="required"
            type="button"
            value="Yes"
          />
          <input
            className="btn btn-default"
            id="us-crimes-1"
            name="us-crimes"
            onClick={[Function]}
            required="required"
            type="button"
            value="No"
          />
        </div>
      </div>
    </div>
    <div
      className="col-sm-12 parentButton"
    >
      <input
        className="btn btn-primary"
        id="submit"
        type="submit"
        value="View Your Eligibility"
      />
    </div>
  </div>
</form>
`;

exports[`components/Form should view eligibility if all fields are entered 1`] = `
<form
  id="form"
  onSubmit={[Function]}
>
  <div
    className="row"
  >
    <div
      className="col-lg-11 col-lg-offset-1"
    >
      <h3
        className="subHead"
      >
        Basic Information
      </h3>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-lg-2 col-lg-offset-1"
    >
      <div
        className="select form-group field-state"
      >
        <label
          className="select-label"
          htmlFor="state"
        >
          Where do you live?
          <span
            className="required"
          >
            *
          </span>
        </label>
        <select
          aria-required="true"
          className="form-control"
          defaultValue="Select State"
          id="state"
          name="state"
          onChange={[Function]}
          required="required"
        >
          <option
            disabled={true}
            value="Select State"
          >
            Select State
          </option>
          <option
            value="Alabama"
          >
            Alabama
          </option>
          <option
            value="Alaska"
          >
            Alaska
          </option>
          <option
            value="Arizona"
          >
            Arizona
          </option>
          <option
            value="Arkansas"
          >
            Arkansas
          </option>
          <option
            value="California"
          >
            California
          </option>
          <option
            value="Colorado"
          >
            Colorado
          </option>
          <option
            value="Connecticut"
          >
            Connecticut
          </option>
          <option
            value="Delaware"
          >
            Delaware
          </option>
          <option
            value="District of Columbia"
          >
            District of Columbia
          </option>
          <option
            value="Florida"
          >
            Florida
          </option>
          <option
            value="Georgia"
          >
            Georgia
          </option>
          <option
            value="Hawaii"
          >
            Hawaii
          </option>
          <option
            value="Idaho"
          >
            Idaho
          </option>
          <option
            value="Illinois"
          >
            Illinois
          </option>
          <option
            value="Indiana"
          >
            Indiana
          </option>
          <option
            value="Iowa"
          >
            Iowa
          </option>
          <option
            value="Kansas"
          >
            Kansas
          </option>
          <option
            value="Kentucky"
          >
            Kentucky
          </option>
          <option
            value="Louisiana"
          >
            Louisiana
          </option>
          <option
            value="Maine"
          >
            Maine
          </option>
          <option
            value="Maryland"
          >
            Maryland
          </option>
          <option
            value="Massachusetts"
          >
            Massachusetts
          </option>
          <option
            value="Michigan"
          >
            Michigan
          </option>
          <option
            value="Minnesota"
          >
            Minnesota
          </option>
          <option
            value="Mississippi"
          >
            Mississippi
          </option>
          <option
            value="Missouri"
          >
            Missouri
          </option>
          <option
            value="Montana"
          >
            Montana
          </option>
          <option
            value="Nebraska"
          >
            Nebraska
          </option>
          <option
            value="Nevada"
          >
            Nevada
          </option>
          <option
            value="New Hampshire"
          >
            New Hampshire
          </option>
          <option
            value="New Jersey"
          >
            New Jersey
          </option>
          <option
            value="New Mexico"
          >
            New Mexico
          </option>
          <option
            value="New York"
          >
            New York
          </option>
          <option
            value="North Carolina"
          >
            North Carolina
          </option>
          <option
            value="North Dakota"
          >
            North Dakota
          </option>
          <option
            value="Ohio"
          >
            Ohio
          </option>
          <option
            value="Oklahoma"
          >
            Oklahoma
          </option>
          <option
            value="Oregon"
          >
            Oregon
          </option>
          <option
            value="Pennsylvania"
          >
            Pennsylvania
          </option>
          <option
            value="Rhode Island"
          >
            Rhode Island
          </option>
          <option
            value="South Carolina"
          >
            South Carolina
          </option>
          <option
            value="South Dakota"
          >
            South Dakota
          </option>
          <option
            value="Tennessee"
          >
            Tennessee
          </option>
          <option
            value="Texas"
          >
            Texas
          </option>
          <option
            value="Utah"
          >
            Utah
          </option>
          <option
            value="Vermont"
          >
            Vermont
          </option>
          <option
            value="Virginia"
          >
            Virginia
          </option>
          <option
            value="Washington"
          >
            Washington
          </option>
          <option
            value="West Virginia"
          >
            West Virginia
          </option>
          <option
            value="Wisconsin"
          >
            Wisconsin
          </option>
          <option
            value="Wyoming"
          >
            Wyoming
          </option>
        </select>
      </div>
    </div>
    <div
      className="col-lg-2"
    >
      <div
        className="radio-group form-group field-us-citizen"
      >
        <label
          className="radio-group-label"
          htmlFor="us-citizen"
        >
          Are you a U.S. Citizen?
          <span
            className="required"
          >
            *
          </span>
        </label>
        <br />
        <div
          aria-label="..."
          className="btn-group"
          role="group"
        >
          <input
            className="btn btn-default"
            id="us-citizen-0"
            name="us-citizen"
            onClick={[Function]}
            required="required"
            type="button"
            value="Yes"
          />
          <input
            className="btn btn-default"
            id="us-citizen-1"
            name="us-citizen"
            onClick={[Function]}
            required="required"
            type="button"
            value="No"
          />
        </div>
      </div>
    </div>
    <div
      className="col-lg-2"
    >
      <div
        className="date form-group field-birthdate"
      >
        <label
          className="date-label"
          htmlFor="birthdate"
        >
          Birthday
          <span
            className="required"
          >
            *
          </span>
        </label>
        <input
          aria-required="true"
          className="form-control"
          id="birthdate"
          name="birthdate"
          onChange={[Function]}
          required="required"
          type="date"
        />
      </div>
    </div>
    <div
      className="col-lg-5"
    >
      <div
        className="radio-group form-group field-us-citizen"
      >
        <label
          className="radio-group-label"
          htmlFor="us-citizen"
        >
          Have you ever been charged, or convicted of a crime?
          <span
            className="required"
          >
            *
          </span>
        </label>
        <br />
        <div
          aria-label="..."
          className="btn-group"
          role="group"
        >
          <input
            className="btn btn-default"
            id="us-crimes-0"
            name="us-crimes"
            onClick={[Function]}
            required="required"
            type="button"
            value="Yes"
          />
          <input
            className="btn btn-default"
            id="us-crimes-1"
            name="us-crimes"
            onClick={[Function]}
            required="required"
            type="button"
            value="No"
          />
        </div>
      </div>
    </div>
    <div
      className="col-sm-12 parentButton"
    >
      <input
        className="btn btn-primary"
        id="submit"
        type="submit"
        value="View Your Eligibility"
      />
    </div>
  </div>
</form>
`;
